tests/foreman/maintain/test_backup_restore.py

```diff
@@ -421,17 +488,24 @@ def test_positive_backup_restore(
     assert 'FAIL' not in result.stdout
 
     # Check for expected files
-    backup_dir = re.findall(fr'{subdir}\/satellite-backup-.*-[0-5][0-9]', result.stdout)[0]
+    backup_dir = re.findall(fr'{subdir}\/{instance}-backup-.*-[0-5][0-9]', result.stdout)[0]
     files = sat_maintain.execute(f'ls -a {backup_dir}').stdout.split('\n')
     files = [i for i in files if not re.compile(r'^\.*$').search(i)]
 
-    if sat_maintain.is_remote_db():
-        expected_files = BASIC_FILES | REMOTE_SAT_FILES
+    if type(sat_maintain) is Satellite:
+        if sat_maintain.is_remote_db():
+            expected_files = BASIC_FILES | REMOTE_SAT_FILES
+        else:
+            expected_files = (
+                BASIC_FILES | OFFLINE_FILES
+                if backup_type == 'offline'
+                else BASIC_FILES | ONLINE_SAT_FILES
+            )
```

Comment:
Could we make this a helper function? Looks like this logic is repeated a couple times.

Reply:
Sure thing :+1: